### ISSUE TITLE
Utility functions meant for external use need a section in the docs

### ISSUE THREAD

**Issue by brittonsmith** [yt-project/yt#2015]
There is currently no place in the narrative docs for the various utility functions that could potentially be externally useful. For example, a good number of the functions in `yt/utilities/math_utils.py` seem to be intended for this type of use. Some of these things are listed in the API docs, but this does not make them very discoverable.

**Comment by ngoldbaum** [comment]
I don't think we should publicly advertise these. They're not in the docs because they're not part of the public API.

In fact I'd be in favor of explicitly making them private, e.g. by renaming `yt.utilities` to `yt._utilities`. If people use them then if we want to change how they work later for some reason we might unintentionally break people's workflows.

**Comment by brittonsmith** [comment]
As it stands, there are a number of functions in `yt/utilities` that aren't used within yt itself. This suggests to me that they were intended for external use.  Either that, or they've fallen out of use in yt and should be removed.

Perhaps it's worth making an explicit division between public and private utilities. I would agree that most of yt's utilities functions are internal and should not be in the public API, but my feeling is that there is likely a non-zero number of exceptions. For example, I was surprised to learn that `ortho_find`, 'quartiles`, and `periodic_position` are all imported by `yt/__init__.py`. I am for identifying those and increasing their visibility, but I also think adopting a `_utilities` for private functionality is good.

**Comment by ngoldbaum** [comment]
OK, I'm fine with having an explicitly enumerated subset of these functions that we expose publicly. Since most of them are pure python we could also think about packaging them separately and publishing on pypi (but don't do that unless you think it's worth doing).

**Comment by brittonsmith** [comment]
I'm open to this perhaps down the road after the functionality in question has all been identified, reorganized, and documented, and that set has stabilized. Setting up and managing an additional package also comes with non-zero effort and headaches. It's not clear to me right now that we as a community have a model for maintaining so many separate packages.

**Comment by chummels** [comment]
For reference, when I wrote `ortho_find()`, I meant for it to be publicly accessible and useable, but was told that `math_utils` was the place to put it, so it has gone mostly under the radar for the last ~5 years.  But I do think it is very useful for a lot of workflows.  I'm in favor of having it, and other functions like it, more visible to users.

**Comment by MatthewTurk** [comment]
@chummels Just to make a brief note, when @brittonsmith and I were talking through this, we both commented on what an exemplar of good documentation and error checking that `ortho_find` was.  Thank you for that effort.
